video/out/vo_gpu_next.c

```diff
@@ -1101,6 +1101,12 @@ static bool draw_frame(struct vo *vo, struct vo_frame *frame)
         target_csp.hdr.max_luma = 0;
         target_csp.hdr.max_cll = 0;
         target_csp.hdr.max_fall = 0;
+
+        // For min_luma specifically, if the display reports a value below the
+        // sRGB reference display minimum luminance, assume it's an OLED display
+        // and set min_luma to 0. Otherwise, leave it to the reference display
+        // value which shouldn't be too different from the reported value.
+        target_csp.hdr.min_luma = (target_csp.hdr.min_luma < 0.2f) ? 0 : 0.2f;
```

Comment:
0 is undefined, you want to use `PL_COLOR_HDR_BLACK`.

Also for cutoff point you can use `(PL_COLOR_SDR_WHITE / PL_COLOR_SDR_CONTRAST)`

I don't know what are the good values here though. I don't see how changing this to binary selection helps. It's quite opinionated decision to categorize all displays into two groups, ignoring the reported luminance.

Reply:
Perhaps the cutoff being 0.1 being makes more sense? The point is to have better out of box default for OLED screens, where min_luma=0.01 raises blacks and users report target-contrast=inf is the correct output.